### PR TITLE
[Purview - API Scanning] API Review 2022-07-01-preview 

### LINKED ISSUE #19807: [Purview - API Scanning] API Review 2022-07-02-preview
[Purview - API Scanning] API Review

PR #19797 
Planning on reservice this time slot.
I will send a meeting invite to azureapirbcore and azure-restapi-int if you don’t see any issues with the time slots. 
Other information regarding the PR/Service
- Service name: Purview-Scanning
- Key Contact(s): @Phani Kumar Vadali, @Cala Zubair
- New or Existing API: New API endpoints to an existing service
- Deployment: Service is already deployed, and these new endpoints are in use in production from UX 
- This change facilitates UX and customers to execute CRUD operations for IntegrationRuntimes and ManagedVirtualNetworks. Right now, UI calls a different Azure service (Azure Data Factory) for executing these operations. Switching over to calling Purview-Scanning will allow us to intercept request/response routing to Azure Data Factory and return Purview specific models/json-responses to the caller 
- Link to PR: [https://github.com/Azure/azure-rest-api-specs/pull/19797](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FAzure%2Fazure-rest-api-specs%2Fpull%2F19797&data=05%7C01%7Cmarkweitzel%40microsoft.com%7Cfbc88fd62cb943c8f86e08da68dbf434%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637937591152696612%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=YmIwV9x2XoLcu8cMrkwVA08m5IbpUKS%2BJ5ROmQ1w%2FT0%3D&reserved=0)
Please let me know if anything else is missing.

### API REVIEWS
---
- [x] [Review session 27-07-2022](https://github.com/Azure/azure-rest-api-specs/issues/19807#issuecomment-1205475130)
- [ ] [Review session 18-08-2022]()

### PR DESCRIPTION
**Pull request by pvadali-az**
# Data Plane API - Pull Request

## API Info: The Basics
Most of the information about your service should be captured in the issue that serves as your [*engagement record*](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/271/Azure-REST-API-Stewardship?anchor=rest-api-stewardship-process).

* Link to engagement record issue:

Is this review for (select one):

- [ ] a private preview
- [x] a public preview
- [ ] GA release

### Change Scope
<sup>This section will help us focus on the specific parts of your API that are new or have been modified. <br/>Please share a link to the design document for the new APIs, a link to the previous Open API document (swagger) if applicable, and the root paths that have been updated. </sup>
* Design Document:
* Previous Open API Doc:
* Updated paths:

## ❔Got questions? Need additional info?? We are here to help!

<details>
  <summary> Contact us!</summary>

The [Azure API Review Board](https://aka.ms/azapi) is dedicated to helping you create amazing APIs. You can read about our mission and learn more about our process on our [wiki](https://aka.ms/azapi).
* 💬 [Teams Channel](https://teams.microsoft.com/l/channel/19%3a3ebb18fded0e47938f998e196a52952f%40thread.tacv2/General?groupId=1a10b50c-e870-4fe0-8483-bf5542a8d2d8&tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47)
* 💌 [email](mailto://azureapirbcore@microsoft.com)

</details>

<details>
  <summary>Click here for links to tools, specs, guidelines & other good stuff</summary>

### Tooling
 * [Open API validation tools](https://aka.ms/swaggertools) were run on this PR. Go here to see [how to fix errors](https://aka.ms/ci-fix)
 * [Spectral Linting](https://aka.ms/style)
 * [Open API Hub](https://aka.ms/openapiportal)

### Guidelines & Specifications
 * [Azure REST API Guidelines](https://aka.ms/guidelines)
 * [OpenAPI Style Guidelines](https://aka.ms/style)
 * [Azure Breaking Change Policy](https://aka.ms/AzBreakingChangesPolicy)

### Helpful Links
 * [Azure DevTools Wiki](https://aka.ms/azapi)

</details>

### CODE REVIEW

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (JiaxingSong) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3620]: preventDataExfiltration should not expose to customers

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (JiaxingSong) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1467]: Today creating multiple managed vnet is not be allowed? The API implementation allows to create vnets other than 'default'

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (JiaxingSong) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3532]: IntegrationRuntimeStatus contains 'kind' only, but no states like 'running', 'failed', 'limited' ？

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1467]: Since this doesn't change the API signature itself and is limited to service side change, I'm resolving this conversation here and make the change in Scanning service. Please feel free to reopen for any other questions/concerns.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3547]: Had offline conversation, Status field comes from specific kind of IntegrationRuntimeStatus

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: suppressed preventDataExfiltration

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1471]: Changed this to enum

**Comment by mikekistler** [comment]
- Add a global security requirement

- Add x-ms-error-code response header on error responses (needs service changes to support)

- All parameters need descriptions

- Operations with a 202 response should specify `x-ms-long-running-operation: true`.

- A 202 response should include an Operation-Location response header.

- Operation might be pageable. Consider adding the x-ms-pageable extension.

- Use camelcase.  E.g. "VNet" -> "Vnet"

- Use consistent path parameter names:
    - Inconsistent parameter names "dataSourceName" and "dataSourceType" for path segment "datasources".
    - Inconsistent parameter names "classificationRuleVersion" and "version" for path segment "versions".

- No need for an allOf when a simple $ref will do. There are many instances of this problem.

- Many body parameters are not marked as required. This is a common mistake. Please double check all body parameters and mark as required unless the body is truly optional.

- The final path parameter on a "put" should specify a maximum length (maxLength) and characters allowed (pattern).

- There are some schemas that specify additionalProperties with type object. This is a common error - please double check.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1076]: 204 should not have a schema in response?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:6017]: Many of them should be a direct ref to the type, not an anonymous type with "allOf".

Swagger appears to be generated?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1155]: expect a better description.

e.g. I've no idea why status is that important to worth a dedicated endpoint (instead of included in "/integrationruntimes/{integrationRuntimeName}" resource).

Maybe this is already talked in meeting, but to a user new to your REST API, the description is what they got.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1325]: Endpoint seems not consistent. Should it be ":disableInteractiveQuery" as an action?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1350]: I think it is an `object` without any properties? That is meaningless to user.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1538]: Use a meaningful operationId. They translates to method name in SDK.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1558]: Similarly, use a meaningful name.

Nobody know what PE stand for.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1554]: why this is an enum with "default"?

I thought you can create multiple vnet via "/managedvirtualnetworks/{managedVNetName}" endpoint?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1623]: same, 204 should return no content.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3389]: from diff, it seems "AmazonPostgreSqlRoleARN" is removed?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3590]: these polymorphism without any meaningful shared property is painful.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3648]: you probably need a reason for having "count"?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3687]: unnecessary "allOf"?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3704]: Should it be enum?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1109]: Path parameter should specify a maximum length (maxLength) and characters allowed (pattern).

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1155]: Agree that this seems unnecessary. Why not just include status in the a get on the resource?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1263]: This body parameter is not marked "required: true". That is a common oversight. How does this operation behave when no body is passed?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1396]: Here is another body parameter that is not "required: true". Is this an oversight?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1472]: A put operation is a create or replace:
```suggestion
        "description": "Creates or Replaces a ManagedVirtualNetwork",
        "operationId": "ManagedVirtualNetworks_CreateOrReplace",
```

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1644]: Put is create or replace
```suggestion
        "description": "Create or replace a ManagedPrivateEndpoint",
        "operationId": "ManagedVirtualNetworks_CreateOrReplacePE",
```

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (mikekistler) [inline specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1492]: Why is this defined as an enum with only one value? Do you expect to allow additional values later? Will these be fixed values or will the user be able to assign their own values? I am concerned that we many not be able to evolve this pattern compatibly.

**Review by mikekistler — CHANGES_REQUESTED** [review]
@weidongxu-microsoft has pointed out a number of issues and I have added a few more.

Please address these and then flag us to re-review.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1492]: Right now UI doesn't have a way to create a ManagedVirtualNetwork. It implicitly calls service with the name "default" while creating ManagedIntegrationRuntime, I think. @JiaxingSong, please provide more context here if I'm missing anything. 


Since we are not letting users create a ManagedVirtualNetwork with any other name besides "default", I had to make this an enum. I'm not sure if there are plans to let users create multiple VirtualNetworks with names of their own choice.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3389]: Yes, `AmazonPostgreSqlRoleARN` is removed as part of a schema migration

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1554]: I was told that we cannot create multiple virtual networks, and that the only virtualnetwork that can be created is forced to be of the name "default"
cc: @JiaxingSong

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1146]: From what I saw in Scanning and DataFactory, `IntegrationRuntime` object is vastly different from `IntegrationRuntimeStatus`. Status is periodically polled by the browser where as IntegrationRuntime_Get is not. This might be why DataFactory had same pattern and a separate endpoint. UI was also calling this the same way and I didn't want to introduce a new change when they switched over from calling DataFactory to calling Scan. 

Let me know if that answers the question.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:5944]: Yes, this is autogenerated. Also, explained in a separate email.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1348]: Explained in the email. This and http 204 have same underlying problem.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: changed all PE occurrences to PrivateEndpoint

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: Changed

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1573]: Discussed in the email

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3575]: PageableResult is a generic object and will change all existing pageable results. Keeping it for now, Scan team will have to change PageableResult and make sure UI experience is not broken for other APIs

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3614]: explained above regarding autogeneration

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3631]: Yes, but there is no clarity on what the expected values are even within my change. Several properties could be enums, but DataFactory's swagger did not mark them as enums and hence swashbuckle did not generate enums from Datafactory provided models. This will be a bigger change and I felt it is a bit out of scope for this PR. Let me know. (also explained in the email)

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: changed

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: Added

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: changed

Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: The operation returns a http 500 when no body is passed

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: Yes, the body is required in this case too. This is an oversight by me while creating external models. Changed.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: changed

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: changed

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (JiaxingSong) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: Today we allow only 'default' as virtual network name, this is because managed vnet runtime feature allows only one vnet for each purview account. In the future we probably will allow user to create more vnet with different names, so @pvadali-az using a enum here will block the flexibility in the coming feature evolvements, I'd prefer to allow arbitrary string as the name and throw 400 from service side if the input name is other than 'default'.   In this way, we could easily allow more values from service implementation instead of release new SDK version

**Review by mikekistler — APPROVED** [review]
Looks good for preview. 👍

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3517]: The `IntegrationRuntime` is added with your PR. I don't see why improving it would disturb any other external models that not in your PR.

That said, I am fine with it as there is already too many such polymorphism in scanning.

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (weidongxu-microsoft) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:1348]: It is used in 200.

So it is intended to be 200 but still with response body as "{}"?

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json]: changed to `/integrationruntimes/{integrationRuntimeName}:disableInteractiveQuery`

File: specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json
Comment (pvadali-az) [reply specification/purview/data-plane/Azure.Analytics.Purview.Scanning/preview/2022-07-01-preview/scanningService.json:3517]: It wouldn't disturb other models, but relying on `Kind` is used at a BaseAdapter level and we have to write some other adapter and conversion flow from ground up for this alone. Since there is a root in the source code where this change can start from and spread across entire repo, I felt this can be postponed until then.

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by tjprescott — APPROVED** [review]

**Comment by tjprescott** [comment]
@pvadali-az you will need to address the failing Avocado checks to merge this.

**Comment by pvadali-az** [comment]
> @pvadali-az you will need to address the failing Avocado checks to merge this.

@tjprescott I fixed avocado errors just now. I will have to wait until next to merge this since service change hasn't rolled out to all regions.

**Pull request merged**
